x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/TransformIndexTemplateRegistryTests.java

```diff
@@ -0,0 +1,180 @@
+        when(adminClient.indices()).thenReturn(indicesAdminClient);
+        when(client.admin()).thenReturn(adminClient);
+        doAnswer(withResponse(new AcknowledgedResponse(true))).when(indicesAdminClient).putTemplate(any(), any());
+
+        clusterService = mock(ClusterService.class);
+
+        List<NamedXContentRegistry.Entry> entries = new ArrayList<>(ClusterModule.getNamedXWriteables());
+        entries.add(new NamedXContentRegistry.Entry(LifecycleAction.class, new ParseField(RolloverAction.NAME), RolloverAction::parse));
+        xContentRegistry = new NamedXContentRegistry(entries);
+
+        putIndexTemplateRequestCaptor = ArgumentCaptor.forClass(PutIndexTemplateRequest.class);
+    }
+
+    public void testTemplates() {
+        TransformIndexTemplateRegistry registry =
+            new TransformIndexTemplateRegistry(Settings.EMPTY, clusterService, threadPool, client, xContentRegistry);
+
+        registry.clusterChanged(createClusterChangedEvent(nodes));
+
+        verify(client.admin().indices(), times(2)).putTemplate(putIndexTemplateRequestCaptor.capture(), anyObject());
+
+        PutIndexTemplateRequest internal = putIndexTemplateRequestCaptor.getAllValues().stream()
+            .filter(r -> r.name().equals(TransformInternalIndexConstants.LATEST_INDEX_VERSIONED_NAME))
+            .findFirst()
+            .orElseThrow(() -> new AssertionError("expected the transform internal index template to be put"));
+
+        assertEquals(Version.CURRENT.id, internal.version().intValue());
+        assertThat(internal.patterns(), Matchers.contains(TransformInternalIndexConstants.LATEST_INDEX_VERSIONED_NAME));
+        assertThat(internal.settings().get("index.number_of_shards"), equalTo("1"));
+        assertThat(internal.settings().get("index.auto_expand_replicas"), equalTo("0-1"));
+        assertThat(internal.settings().size(), equalTo(2));
+
+        PutIndexTemplateRequest audit = putIndexTemplateRequestCaptor.getAllValues().stream()
+            .filter(r -> r.name().equals(TransformInternalIndexConstants.AUDIT_INDEX))
+            .findFirst()
+            .orElseThrow(() -> new AssertionError("expected the transform audit index template to be put"));
+
+        assertEquals(Version.CURRENT.id, audit.version().intValue());
+        assertThat(audit.patterns(), Matchers.contains(TransformInternalIndexConstants.AUDIT_INDEX_PREFIX + "*"));
+        Alias alias = audit.aliases().iterator().next();
+        assertThat(alias.name(), equalTo(".transform-notifications-read"));
+        assertTrue(alias.isHidden());
+        assertThat(audit.settings().get("index.number_of_shards"), equalTo("1"));
+        assertThat(audit.settings().get("index.auto_expand_replicas"), equalTo("0-1"));
+        assertThat(audit.settings().get("index.hidden"), equalTo("true"));
+        assertThat(audit.settings().size(), equalTo(3));
+    }
+
+    public void testRefactoredMappingsAreSameAsOld() throws IOException {
```

Comment:
This test is key because it proves the mappings have not been altered by the refactoring. The above test asserts none of the other settings have changed.

The mappings defined in `TransformInternalIndex` are used in this test and no longer required once I've proved the mappings have not changed. Once the build is green and the PR approved I'd like to remove the static mapping definitions in `TransformInternalIndex` and this test.